### PR TITLE
Better mimic of RSquare sklearn implementation

### PR DESCRIPTION
**Pull request by lum4chi**
# Description

Brief Description of the PR:
I notice that r2_score and RSquare works differently in managing "-inf":
```python
import tensorflow as tf
from tensorflow_addons.metrics import RSquare
from sklearn.metrics import r2_score

actuals = tf.constant([0, 0], dtype=tf.float32)
preds = tf.constant([1e-3, 1e-3], dtype=tf.float32)
actuals = tf.cast(actuals, dtype=tf.float32)
preds = tf.cast(preds, dtype=tf.float32)

r2_score(actuals, preds)  # return 0.0

m = RSquare()
m.update_state(actuals, preds)
m.result().numpy()  # return -inf
```

Fixes # (issue)
I've replaced "-inf" in `raw_scores` with 0 to comply with sklearn implementation.

## Type of change

- [x] Bug fix
- [ ] New Tutorial
- [ ] Updated or additional documentation
- [x] Additional Testing
- [ ] New Activation and the changes conform to the [activation contribution guidelines](https://github.com/tensorflow/addons/blob/master/tensorflow_addons/activations/README.md#contribution-guidelines)
- [ ] New Callback and the changes conform to the [callback contribution guidelines](https://github.com/tensorflow/addons/blob/master/tensorflow_addons/callbacks/README.md#contribution-guidelines)
- [ ] New Image addition and the changes conform to the [image op contribution guidelines](https://github.com/tensorflow/addons/blob/master/tensorflow_addons/image/README.md#contribution-guidelines)
- [ ] New Layer and the changes conform to the [layer contribution guidelines](https://github.com/tensorflow/addons/blob/master/tensorflow_addons/layers/README.md#contribution-guidelines)
- [ ] New Loss and the changes conform to the [loss contribution guidelines](https://github.com/tensorflow/addons/blob/master/tensorflow_addons/losses/README.md#contribution-guidelines)
- [ ] New Metric and the changes conform to the [metric contribution guidelines](https://github.com/tensorflow/addons/blob/master/tensorflow_addons/metrics/README.md#contribution-guidelines)
- [ ] New Optimizer and the changes conform to the [optimizer contribution guidelines](https://github.com/tensorflow/addons/blob/master/tensorflow_addons/optimizers/README.md#contribution-guidelines)
- [ ] New RNN Cell and the changes conform to the [rnn contribution guidelines](https://github.com/tensorflow/addons/blob/master/tensorflow_addons/rnn/README.md#contribution-guidelines)
- [ ] New Seq2seq addition and the changes conform to the [seq2seq contribution guidelines](https://github.com/tensorflow/addons/blob/master/tensorflow_addons/seq2seq/README.md#contribution-guidelines)
- [ ] New Text addition and the changes conform to the [text op contribution guidelines](https://github.com/tensorflow/addons/blob/master/tensorflow_addons/text/README.md#contribution-guidelines)

# Checklist:

- [ ] I've properly [formatted my code according to the guidelines](https://github.com/tensorflow/addons/blob/master/CONTRIBUTING.md#coding-style)
    - [x] By running Black + Flake8
    - [ ] By running pre-commit hooks
- [ ] This PR addresses an already submitted issue for TensorFlow Addons
- [ ] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] This PR contains modifications to C++ custom-ops

# How Has This Been Tested?

If you're adding a bugfix or new feature please describe the tests that you ran to verify your changes:
*

### CODE REVIEW

**Comment by lum4chi** [comment]
@googlebot I signed it!

File: tensorflow_addons/metrics/r_square.py [hunk 1]
```diff
@@ -130,6 +130,9 @@ def result(self) -> tf.Tensor:
         mean = self.sum / self.count
         total = self.squared_sum - self.sum * mean
         raw_scores = 1 - (self.res / total)
+        raw_scores = tf.where(
```
Comment (bhack) [inline tensorflow_addons/metrics/r_square.py:133]: Probably you could use
```
 raw_scores = tf.where(tf.math.is_inf(raw_scores),
                      tf.zeros_like(raw_scores), raw_scores)
```

Comment (lum4chi) [reply tensorflow_addons/metrics/r_square.py:133]: You're right, thanks!

File: tensorflow_addons/metrics/r_square.py [hunk 2]
```diff
@@ -130,6 +130,9 @@ def result(self) -> tf.Tensor:
...
+            tf.math.is_inf(raw_scores), tf.zeros_like(raw_scores), raw_scores
```
Comment (WindQAQ) [inline tensorflow_addons/metrics/r_square.py:134]: `tf.where`  supports broadcast. We can use `0` instead of  `tf.zeros_like(raw_scores)` to avoid materialize larger constant.

**Review by WindQAQ — CHANGES_REQUESTED** [review]

File: tensorflow_addons/metrics/r_square.py (hunk 2 above)
Comment (bhack) [reply tensorflow_addons/metrics/r_square.py:134]: Yes I don't know if probably `tf.where(tf.math.is_inf(raw_scores),0, raw_scores)` could be ok.

Comment (lum4chi) [reply tensorflow_addons/metrics/r_square.py]: Clever!

Comment (lum4chi) [reply tensorflow_addons/metrics/r_square.py]: Uhm, this change appears to break some tests...

Comment (bhack) [reply tensorflow_addons/metrics/r_square.py]: You need `0.`

**Review by bhack — APPROVED** [review]

**Review by WindQAQ — APPROVED** [review]
